### PR TITLE
[plugin.video.zdf_de_2016] 1.0.1

### PR DESCRIPTION
**Pull request by generia**
##### Kodi Addon "ZDF Mediathek 2016"
ZDF is a German public television channel providing it's television shows online in their so called "ZDF Mediathek".

This is yet another version of the "ZDF Mediathek", since the existing addons stopped working due to the website relaunch end of October 2016 and updates of the existing plugins were not yet available my wife forced me to take action.

*Features*
- combined search function over multiple result pages
- optimized for fast HTML parsing of the zdf.de website
- search history
- rubric browsing
- live TV

*Compatibility*
This addon was tested with 
- Kodi 16.1 "Jarvis", Confluence Skin, MacOS
- Kode 17.0 "Krypton" beta 5, Estouchy Skin, MacOS/Linux 

### Checklist:

- [X] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [X] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [X] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practice but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-plugins/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

### CODE REVIEW

File: plugin.video.zdf_de_2016/addon.py [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+        #li.addContextMenuItems([('Suche -AK','xbmc.RunPlugin(%s?nix=da)'%(self.baseUrl))])
+        #li.addContextMenuItems(['Item-Menu', 'RunPlugin(plugin://'+ self.handle +'/'])
+        #self.context.log.info("[Response] - {} -> {}, title='{}' date='{}'", item.isFolder, url, title, date)
+        xbmcplugin.addDirectoryItem(handle=self.handle, url=url, listitem=li, isFolder=item.isFolder)
+    
+    def close(self):
+        xbmcplugin.endOfDirectory(self.handle)
+        
+    def sendError(self, message, action=Action()):
+        self._sendMessage(xbmcgui.NOTIFICATION_ERROR, "Error", message, action)
```
Comment (phil65) [inline plugin.video.zdf_de_2016/addon.py]: Would be cool to translate these for next update.

File: plugin.video.zdf_de_2016/de/generia/kodi/plugin/frontend/zdf/search/SearchHistory.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        
+        if contentTypes == 'None':
+            contentTypes = None
+        #xbmc.log("_parseEntry: " + str(date) + "-" + query, level=xbmc.LOGNOTICE)
+        return HistoryEntry(query, contentTypes, date)  
+
+    def _saveEntries(self, entries):
+        #xbmc.log("_saveEntries: " + str(len(entries)), level=xbmc.LOGNOTICE)
+                 
+        file = open(self.storeFile, "w")
```
Comment (phil65) [inline plugin.video.zdf_de_2016/de/generia/kodi/plugin/frontend/zdf/search/SearchHistory.py]: "with" statement / try-except for IO, please.

**Comment by generia** [comment]
I cleaned up the issues. 
Changes are forced pushed to this pull request.

File: plugin.video.zdf_de_2016/addon.xml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+	<disclaimer lang="en">Some parts of this addon may not be legal in your country of residence - please check with your local laws before installing.</disclaimer>
+	<license>GNU GENERAL PUBLIC LICENSE. Version 2, June 1991</license>
+	<source>https://github.com/generia/plugin.video.zdf_de_2016</source>
+	<forum></forum>
+	<email>alex at generia dot de</email>
+	<website>http://www.zdf.de</website>
+    <platform>all</platform>
+    <news>
+1.0.1 (2016-12-03)
+- added rubrics: Live-TV, Shows A-Z
```
Comment (phil65) [inline plugin.video.zdf_de_2016/addon.xml]: I think you need to use [CR] for line breaks.

**Comment by phil65** [comment]
looks good to me now, apart from that last comment. Thx!

File: plugin.video.zdf_de_2016/addon.xml (hunk 3 above)
Comment (Razzeee) [inline plugin.video.zdf_de_2016/addon.xml]: Shouldn't be a problem. Or I need to change this for my addons two. ;)

Comment (generia) [inline plugin.video.zdf_de_2016/addon.xml]: Just retested this version on my Raspberry. Works fine. Is the CR issue a blocker?

PS: Background:
I normal develop on OSX. When I checkout the repo-plugins Jarvis branch git shows a bunch of files as modified. These modifications are only due to cr/lf lineendings which git is handling during checkout. There are lots of posts how to remove this, but none actually work for me.

As a last resort I was using an old Windows machine to push the plugin-source (using autocrlf=true).

Comment (phil65) [inline plugin.video.zdf_de_2016/addon.xml]: nope, no blocker. didnt know that this worked fine.

**Comment by MartijnKaijser** [comment]
@generia there's an issue with the addon.xml with wrong defination of the <icon> wich does not exist. I fixed it in in our repo but please fix it yourself as well.

**Comment by generia** [comment]
@MartijnKaijser thx for the info, I will pull your changes to my repo.

**Pull request merged**